### ISSUE TITLE
Cannot install Taskcat

### ISSUE THREAD

**Issue by jmcguire23** [aws-quickstart/taskcat#16]
I am trying to install Taskcat on my Macbook, but I am getting this error each time I try to run the installation script.

    Traceback (most recent call last):
    File "/usr/local/bin/taskcat", line 13, in <module>
    from taskcat import TaskCat
    ImportError: No module named taskcat

The instructions in the readme for installing do not work, and I have had to try and curl the pip3-install-master file manually and run using my python install.

**Comment by avattathil** [comment]
Thank you reporting this! 
Do you have python3 on your MacBook?

**Comment by avattathil** [comment]
@jmcguire23  I was able to recreate this issue by setting my default python back to python v2
Installing the module using `pip2` should solve the issue are seeing. Please give it a try. 
If that indeed fixes this issue I will update the installer.
**`pip2 install taskcat --no-cache-dir --upgrade`**
or
**`pip install taskcat --no-cache-dir --upgrade`**

**Comment by jmcguire23** [comment]
Hmm trying this I can see taskcat under my `/usr/local/bin` directory now, as previously this wouldn't appear, however I am still getting the same error.

**Comment by avattathil** [comment]
what is the output of `pip show taskcat`

**Comment by jmcguire23** [comment]
`pip show taskcat` returns the following:

```
Name: taskcat
Version: 2017.615.201742
Summary: An OpenSource Cloudformation Deployment Framework
Home-page: https://aws-quickstart.github.io/taskcat/
Author: Tony Vattathil, Santiago Cardenas, Shivansh Singh
Author-email: tonynv@amazon.com, sancard@amazon.com, sshvans@amazon.com
License: Apache License 2.0
Location: /usr/local/lib/python3.6/site-packages
Requires: boto3, pyfiglet, pyyaml, tabulate, yattag
```

and `python --version` returns `python 2.7.10`

**Comment by avattathil** [comment]
Ok, so you should have the module installed now try: 
`curl https://raw.githubusercontent.com/aws-quickstart/taskcat/master/taskcat_test.py -o taskcat && chmod 755 taskcat`
then 
`./taskcat`

If you are still having trouble maybe the docker version would be a better option? 
Step 1. Install stable docker version from here:  [https://docs.docker.com/docker-for-mac/install/](https://docs.docker.com/docker-for-mac/install/)
Step 2.`curl -s https://raw.githubusercontent.com/aws-quickstart/taskcat/master/installer/docker-install-master| sudo python -E`

Please do let me know that it solves your issues.

**Comment by jmcguire23** [comment]
Ok, I think installing the docker version is working. I am running into another issue when I run my test I get `botocore.exceptions.ClientError: An error occurred (IllegalLocationConstraintException) when calling the CreateBucket operation: The unspecified location constraint is incompatible for the region specific endpoint this request was sent to.`

**Comment by avattathil** [comment]
Try setting your default region to`us-east-1`
If you are using a boto profile set the default to `us-east-1`
[profile default]
output = json
region = us-east-1

**Comment by jmcguire23** [comment]
Seems whatever region I put in the config.yml I get an error. 

My new error looks like this:

`Hint: The name specfied as value of qsname (quickstart-ibm-mq)
                    must match the root directory of your project`

**Comment by avattathil** [comment]
Change the value of `qsname:` from `ibm-mq-aws-quickstart` to the `quickstart-ibm-mq` (git repo name) The value of qsname has to match the folder name, taskcat looks for this folder in your pwd. 

So if you want to kick off the test cd to the directory where you cloned the repo. 
from there make sure your the `config.yml` exists
ls -l `quickstart-ibm-mq/ci/config.yml` 
**from the same dir** run:
`taskcat -c quickstart-ibm-mq/ci/config.yml` 

I will add a section to the docs page with the errors encountered to provide better guidance and also make a few updates in the next version.

**Comment by jmcguire23** [comment]
I have updated taskcat and now I get the error which you detail in your FAQ page.
```
botocore.exceptions.ClientError: An error occurred (IllegalLocationConstraintException) when calling the CreateBucket operation: The unspecified location constraint is incompatible for the region specific endpoint this request was sent to.
```

Changing the region default in `~/.aws/config` to `us-east-1` does not resolve this, as you have suggested.

**Comment by avattathil** [comment]
Can you paste the logs here so I can see the debug output? 
Cut and paste is fine

**Comment by jmcguire23** [comment]
Here is the full console output.
```
[dockerize]
 _            _             _   
| |_ __ _ ___| | _____ __ _| |_ 
| __/ _` / __| |/ / __/ _` | __|
| || (_| \__ \   < (_| (_| | |_ 
 \__\__,_|___/_|\_\___\__,_|\__|
                                

version 2017.627.23425


[taskcat] :AWS AccountNumber: 	 [299743145002]
[taskcat] :Authenticated via: 	 [role] 
[TASKCAT ] :Reading Config form: ./ci/config.yml


[taskcat] :Reading Config form: ./ci/config.yml
[taskcat] |Queing test => quickstart-ibm-mqt1 


[taskcat] |CONTENTS OF  S3 BUCKET
[INFO   ] :Staging Bucket => taskcat-ibm-mq-aws-quickstart-1adf0349
Traceback (most recent call last):
  File "/usr/local/bin/taskcat", line 54, in <module>
    main()
  File "/usr/local/bin/taskcat", line 43, in main
    tcat_instance.stage_in_s3(taskcat_cfg)
  File "/usr/lib/python3.5/site-packages/taskcat/taskcat.py", line 310, in stage_in_s3
    s3.create_bucket(Bucket=auto_bucket)
  File "/usr/lib/python3.5/site-packages/boto3/resources/factory.py", line 520, in do_action
    response = action(self, *args, **kwargs)
  File "/usr/lib/python3.5/site-packages/boto3/resources/action.py", line 83, in __call__
    response = getattr(parent.meta.client, operation_name)(**params)
  File "/usr/lib/python3.5/site-packages/botocore/client.py", line 310, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/usr/lib/python3.5/site-packages/botocore/client.py", line 599, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (IllegalLocationConstraintException) when calling the CreateBucket operation: The unspecified location constraint is incompatible for the region specific endpoint this request was sent to.
```

**Comment by avattathil** [comment]
Ok, I think I know what is going on (please try this so I can narrow it down)

Create a bucket in **us-east-1** something like -> `taskcat-ibm-mq-stage`
then add this key to your config.yml
[global]
s3bucket: `taskcat-ibm-mq-stage`

One more slight change (during this test)
Make sure to run from outside your repo

cd .. #_to change to the folder where your clone resides_ (assuming that you are in the repo dir from the logs above)

then run:
`taskcat -c sample-project/ci/config.yml`
cli output will:
change from -> `[taskcat] :Reading Config form: ./ci/config.yml`
change to -> `[taskcat] :Reading Config form: sample-project/ci/config.yml`

rerun and let me know if that fixes the issues

**Comment by avattathil** [comment]
@jmcguire23, let me know if you have any further issues.
